src/Stepper/index.js

```diff
@@ -4,79 +4,59 @@ import * as React from "react";
 import StepperStateless from "./StepperStateless";
 import validateIncrement from "../utils/validateIncrement";
 import validateDecrement from "../utils/validateDecrement";
-
-import type { Props, State } from "./index";
-
-class Stepper extends React.PureComponent<Props, State> {
-  state = {
-    value: this.props.defaultValue || 0,
-  };
-
-  setValueAndInjectCallback = (value: number) => {
-    const { onChange } = this.props;
-    if (onChange) {
-      onChange(value);
-    }
-    this.setState({ value });
+import useStateWithCallback from "../hooks/useStateWithCallback";
+
+import type { Props } from "./index";
+
+const Stepper = ({
+  onChange,
+  defaultValue = 0,
+  step = 1,
+  maxValue = Number.POSITIVE_INFINITY,
+  minValue = Number.NEGATIVE_INFINITY,
+  ...props
+}: Props) => {
+  const [value, setValue] = useStateWithCallback<number>(defaultValue, onChange);
+
+  const incrementCounter = () => {
+    setValue(validateIncrement({ value, maxValue, step }));
   };
 
-  incrementCounter = () => {
-    const { value } = this.state;
-    const { maxValue = Number.POSITIVE_INFINITY, step = 1 } = this.props;
-
-    this.setValueAndInjectCallback(validateIncrement({ value, maxValue, step }));
+  const decrementCounter = () => {
+    setValue(validateDecrement({ value, minValue, step }));
   };
 
-  decrementCounter = () => {
-    const { value } = this.state;
-    const { minValue = Number.NEGATIVE_INFINITY, step = 1 } = this.props;
-
-    this.setValueAndInjectCallback(validateDecrement({ value, minValue, step }));
-  };
-
-  handleKeyDown = (ev: SyntheticKeyboardEvent<HTMLInputElement>) => {
+  const handleKeyDown = (ev: SyntheticKeyboardEvent<HTMLInputElement>) => {
     if (ev.keyCode === 40) {
       ev.preventDefault();
-      this.decrementCounter();
+      decrementCounter();
     }
     if (ev.keyCode === 38) {
       ev.preventDefault();
-      this.incrementCounter();
+      incrementCounter();
     }
   };
 
-  render() {
-    const {
-      onBlur,
-      onFocus,
-      disabled,
-      name,
-      dataTest,
-      maxValue,
-      minValue,
-      titleIncrement,
-      titleDecrement,
-    } = this.props;
-    const { value } = this.state;
-    return (
-      <StepperStateless
-        disabled={disabled}
-        dataTest={dataTest}
-        value={value}
-        name={name}
-        minValue={minValue}
-        maxValue={maxValue}
-        onKeyDown={this.handleKeyDown}
-        onBlur={onBlur}
-        onFocus={onFocus}
-        onIncrement={this.incrementCounter}
-        onDecrement={this.decrementCounter}
-        titleIncrement={titleIncrement}
-        titleDecrement={titleDecrement}
-      />
-    );
-  }
-}
+  const { onBlur, onFocus, disabled, name, dataTest, titleIncrement, titleDecrement } = props;
+  return (
+    <StepperStateless
+      disabled={disabled}
+      dataTest={dataTest}
+      value={value}
+      name={name}
+      minValue={minValue}
```

Comment:
Looks good 💪 , only problem is that `minValue` and `maxValue` renders with `Infinity` or `-Infinity` now 😕 

Can you look and fix @mainframev? 🙏

Reply:
fixed